solutions/CF Packmen.mdx

```diff
@@ -0,0 +1,293 @@
+---
+id: cf-packmen
+source: CF
+title: Packmen
+author: Kevin Sheng
+---
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+Binary search on the minimum time it will take for all the Packmen to eat all the pellets.
+For each Packman, we can greedily calculate the maximum number of pellets that it can eat in the given amount of time and then remove those pellets.
+Note that it is optimal for all of the Packmen to eat continuous segments of the food, and all of the segments must be non-intersecting. They must be continuous because if they weren't, a Packman could always eat it while going from one segment to another, thus making it continuous. Also, they must be non-intersecting because if they *were*, two Packmen would target the same pellet, thus resulting in wasted time.
```

Comment:
Maybe phrase it as
"A packman will visit all pellets between the leftmost and rightmost pellets a it eats, so it can eat all of them without any time penalty" ?

Reply:
"Also, they must be non-intersecting because if they *were*, either packman can ignore the pellets of the other packman." might be better?